### PR TITLE
Custom pronouns permissions

### PR DESCRIPTION
**Pull request by belac9001**
this *should* work fine, from testing with pre reg/new at-con order/inline attendee edit, overriding with custom pronoun seemed to work fine. if overriding the preferredPronoun in Attendee.java should be done a different way, then i can change that :") 

#131

### CODE REVIEW

File: src/main/resources/templates/order/orders-id-attendees-id.html [hunk 1]
```diff
@@ -81,6 +81,10 @@
                 <label for="inputPronoun" class="col-form-label-sm">Preferred Pronoun</label>
                 <input type="text" class="form-control" readonly th:field="*{preferredPronoun}" id="inputPronoun">
             </div>
+            <div class="form-group col-md-3" th:if="${#authorization.expression('hasAuthority(''set_custom_pronouns'')')}">
+                <label for="inputCustomPronoun" class="col-form-label-sm">Custom Pronoun</label>
+                <input type="text" th:field="*{customPronoun}" th:errorclass="is-invalid" class="form-control" id="inputCustomPronoun">
```
Comment (jashort) [inline src/main/resources/templates/order/orders-id-attendees-id.html:86]: This should be read-only on pages where you're not editing the record. Just change it from
` id="inputCustomPronoun">`
to 
` id="inputCustomPronoun" readonly>`
should do it

**Comment by jashort** [comment]
It bugs me that if you save an attendee with a custom pronoun, then edit the attendee and save it again, you lose the custom pronoun unless you type it in again.

Option 1 would be to detect if the pronoun is custom, and if it is fill in the custom pronoun box (copy that value back in to Attendee.customPronoun on load)
Option 2 would be to add another column in the database for customPronoun and save it all the way through.

**Comment by belac9001** [comment]
do you have a preference for which option you think should be implemented? 

Option 2 feels nice because i feel like we're properly saving it as it's own thing, but i think some badge printing logic may need to be changed? not completely sure.
for Option 1, could check if the pronoun is in `Attendee.PRONOUNS`, and if not then we display the pronoun in the custom pronoun box

**Comment by jashort** [comment]
Yeah, option 2 feels cleaner and option 1 feels kinda hack-ey to me.
I think the badge creation logic would only have to change a little. AttendeeBadgeDTO.fromAttendee() around line 105 is the function that turns Attendee objects in to the AttendeeBadgeDTO objects that the badge printing system uses. So adding logic there to say if customPronouns not empty, use that, else use attendee.getPreferredPronoun().

**Comment by jashort** [comment]
For saving/loading the new field, I think it'll just be editing the SQL queries and appropriate *RowMapper functions in the repository classes.

**Comment by belac9001** [comment]
added custom_pronoun column

**Review by jashort — APPROVED** [review]
I think the only kinda rough spot is that if a custom pronoun is set, the regular pronouns are still selected. And for regular people editing it, it's not obvious that the custom pronoun is set.

But I don't expect that will come up too often.

**Pull request merged**